tests/core/lowering/test_exception_elimination_pass.cpp

```diff
@@ -169,3 +173,189 @@ TEST(LoweringPasses, EliminateExceptionOrPassPattern_Negative) {
   }
   EXPECT_EQ(1, if_count);
 }
+
+TEST(LoweringPasses, EliminateExceptionsSafeIfBlock) {
```

Comment:
The difference is that the graph will be collapsed since the exception will be removed from the graph. This is the same as the graph effect of the `torch::jit::EliminateExceptions` lowering pass. For instance, the change would be:

**Original Graph:**
```
    graph(%x, %y):
      %dim : int = aten::dim(%x)
      %48 : int = prim::Constant[value=2]()
      %66 : bool = aten::eq(%48, %dim)
      %45 : str = prim::Constant[value="EXCEPTION"]()
      %4 : Tensor = prim::If(%66)
        block0():
          = prim::RaiseException(%45)
          -> (%x)
        block1():
          %res = aten::mul(%x, %y)
          -> (%res)
      return (%4)
```

**New Graph:**
```
    graph(%x : Tensor,
              %y : Tensor):
      %6 : Tensor = aten::mul(%x, %y)
      return (%6)
```